### PR TITLE
Cleaned struct tokens for map_session_data and status_change

### PR DESCRIPTION
**Pull request by Singe-Horizontal**
* **Description of Pull Request**: 
#7356 updated map_session_data and status_change from struct to class specifiers
This PR replaces elaborated type specifier in declarations from struct to class and removes them when needed ( most were C left-overs )

### CODE REVIEW

File: src/map/status.cpp [hunk 1]
```diff
@@ -9299,7 +9299,7 @@ struct status_change *status_get_sc(struct block_list *bl)
  */
 void status_change_init(struct block_list *bl)
 {
-	struct status_change *sc = status_get_sc(bl);
+	status_change *sc = status_get_sc(bl);
 	nullpo_retv(sc);
 	memset(sc, 0, sizeof (struct status_change));
```
Comment (aleos89) [inline src/map/status.cpp:9304]:
```suggestion
	memset(sc, 0, sizeof (status_change));
```

**Review by aleos89 — COMMENTED** [review]
Thanks for all the effort you put into this!

There are two lines I can't review which are throwing a couple warnings, both in mob.cpp:
Line 3662:
```cpp
flag=( md->sc.getSCE(cond2) != NULL );
```
Should be:
```cpp
flag=( md->sc.getSCE(static_cast<int32>(cond2)) != NULL );
```

And line 3773:
```cpp
flag = (md->sc.getSCE(ms[i]->cond2)!=NULL);
```
Should be:
```cpp
flag = (md->sc.getSCE(static_cast<int32>(ms[i]->cond2))!=NULL);
```

File: src/map/status.cpp (hunk 1 above)
Comment (Singe-Horizontal) [reply src/map/status.cpp]: I'm not the person who refactored the statuses, do you want me to do those fixes here?

File: src/map/buyingstore.cpp [hunk 2]
```diff
@@ -702,7 +702,7 @@ void do_init_buyingstore_autotrade( void ) {
 					at->sit = battle_config.feature_autotrade_sit;
 
 				// initialize player
-				CREATE(at->sd, struct map_session_data, 1);
+				CREATE(at->sd, map_session_data, 1);
```
Comment (Lemongrass3110) [inline src/map/buyingstore.cpp:705]:
```suggestion
				CREATE(at->sd, map_session_data, 1); // TODO: Dont use Memory Manager allocation anymore and rely on the C++ container
```

File: src/map/vending.cpp [hunk 3]
```diff
@@ -616,7 +616,7 @@ void do_init_vending_autotrade(void)
 					at->sit = battle_config.feature_autotrade_sit;
 
 				// initialize player
-				CREATE(at->sd, struct map_session_data, 1);
+				CREATE(at->sd, map_session_data, 1);
```
Comment (Lemongrass3110) [inline src/map/vending.cpp:619]:
```suggestion
				CREATE(at->sd, map_session_data, 1); // TODO: Dont use Memory Manager allocation anymore and rely on the C++ container
```

File: src/map/buyingstore.cpp (hunk 2 above)
Comment (Lemongrass3110) [inline src/map/buyingstore.cpp:705]:
```suggestion
				CREATE(at->sd, map_session_data, 1); // TODO: Dont use Memory Manager allocation anymore and rely on the C++ container
```

**Review by Lemongrass3110 — APPROVED** [review]

File: src/map/status.cpp (hunk 1 above)
Comment (aleos89) [reply src/map/status.cpp]: Yeah, it was just a couple other warnings that are popping up from the change so might as well fix it now!

**Pull request merged**